project/src/Agents/GSTSpatialAgent.gd

```diff
@@ -0,0 +1,166 @@
+# A specialized steering agent that updates itself every frame so the user does
+# not have to.
+extends GSTNodeAgent
+class_name GSTSpatialAgent
```

Comment:
Same questions as before:

- Should this be split into specific classes, rather than one with several modes for different kinds of bodies? GSTAgentKinematicBody, GSTAgentRigidBody, GSTAgentSpatial would tell me straight what each is for. And when fixing bugs or extending them it might be simpler to maintain.
- Would you [commonly] use steering behaviors on a rigidbody? If not, the users can use the abstract base type and roll out their own code?

Reply:
I'll split them up. I didn't exactly have a specific reason for joining them into one besides that I just did, and yeah, it would make things clearer.